src/wavefront-obj-parser.js

```diff
@@ -2,47 +2,50 @@ module.exports = ParseWavefrontObj
 
 // Map .obj vertex info line names to our returned property names
 var vertexInfoNameMap = {v: 'vertex', vt: 'uv', vn: 'normal'}
-// The returned properties that we will populate
-var parsedProperties = ['normal', 'uv', 'vertex', 'normalIndex', 'uvIndex', 'vertexIndex']
+
 function ParseWavefrontObj (wavefrontString) {
+  'use strict'
+
   var parsedJSON = {normal: [], uv: [], vertex: [], normalIndex: [], uvIndex: [], vertexIndex: []}
 
   var linesInWavefrontObj = wavefrontString.split('\n')
 
+  var currentLine, currentLineTokens, vertexInfoType, i, k
+
   // Loop through and parse every line in our .obj file
-  linesInWavefrontObj.forEach(function (currentLine) {
+  for (i = 0; i < linesInWavefrontObj.length; i++) {
+    currentLine = linesInWavefrontObj[i]
     // Tokenize our current line
-    var currentLineTokens = currentLine.split(' ')
+    currentLineTokens = currentLine.split(' ')
     // vertex position, vertex texture, or vertex normal
-    var vertexInfoType = vertexInfoNameMap[currentLineTokens[0]]
+    vertexInfoType = vertexInfoNameMap[currentLineTokens[0]]
+
     if (vertexInfoType) {
-      parsedJSON[vertexInfoType] = parsedJSON[vertexInfoType].concat(currentLineTokens.slice(1))
-      return
+      for (k = 1; k < currentLineTokens.length; k++) {
+        parsedJSON[vertexInfoType].push(parseFloat(currentLineTokens[k]))
+      }
+      continue
     }
+
     if (currentLineTokens[0] === 'f') {
       // Get our 4 sets of vertex, uv, and normal indices for this face
-      for (var i = 1; i < 5; i++) {
+      for (k = 1; k < 5; k++) {
         // If there is no fourth face entry then this is specifying a triangle
         // in this case we push `-1`
         // Consumers of this module should check for `-1` before expanding face data
-        if (i === 4 && !currentLineTokens[4]) {
+        if (k === 4 && !currentLineTokens[4]) {
           parsedJSON.vertexIndex.push(-1)
           parsedJSON.uvIndex.push(-1)
           parsedJSON.normalIndex.push(-1)
         } else {
-          var indices = currentLineTokens[i].split('/')
-          parsedJSON.vertexIndex.push(Number(indices[0]) - 1) // We zero index
-          parsedJSON.uvIndex.push(Number(indices[1]) - 1) // our face indices
-          parsedJSON.normalIndex.push(Number(indices[2]) - 1) // by subtracting 1
+          var indices = currentLineTokens[k].split('/')
+          parsedJSON.vertexIndex.push(parseInt(indices[0], 10) - 1) // We zero index
+          parsedJSON.uvIndex.push(parseInt(indices[1], 10) - 1) // our face indices
+          parsedJSON.normalIndex.push(parseInt(indices[2], 10) - 1) // by subtracting 1
         }
       }
     }
-  })
-  // Convert our parsed strings into floats (trims trailing insignificant `0`s)
-  parsedProperties.forEach(function (property) {
```

Comment:
Yes, this accounted for about 70% of the perf improvement (the benchmark was down to ~1550ms at this point). The big issue here (aside from the fact that it can be avoided altogether by using parseFloat directly when needed) is the use of concat which is creating a new array for every values of the initial array. Replacing

```js
return accumulator.concat(parseFloat(nextValue))
```

with

```js
accumulator.push(newValue)
return accumulator
```

was already a significant improvement